### PR TITLE
Expose recents transactions

### PR DESCRIPTION
**Pull request by tharvik**
* keep a list of transactions in a tokio agent
* split between `ThinTransaction` which is handled by sieve/drop and `FullTransaction` which is kept locally (with an added timestamp)
* add an exec test runner, taking every executable file in `tests/` and running it
  * add a test library based on drynx
* found a workaround for `chrono`'s RUSTSEC (ignoring it and add a TODO (it's a workaround, not a fix sadly))

### CODE REVIEW

File: src/bin/server/recent_transactions.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        resp: oneshot::Sender<Vec<FullTransaction>>,
+    },
+}
+
+#[derive(Clone)]
+pub struct RecentTransactions {
+    agent: mpsc::Sender<Commands>,
+}
+
+/// Own the recent transactions
```
Comment (ineiti) [inline src/bin/server/recent_transactions.rs]: That comment doesn't help me. Can you be a bit more explicit?

File: src/bin/server/recent_transactions.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        self.agent
+            .send(Commands::Put {
+                sender,
+                thin,
+                resp: tx,
+            })
+            .await
+            .map_err(|_| Error::GoneOnSend)?;
+
+        rx.await.map_err(|_| Error::GoneOnRecv)
```
Comment (ineiti) [inline src/bin/server/recent_transactions.rs:62]: Is that some kind of `await` with channels? If yes, can you replace this with a 'real' `async`?

File: src/bin/server/recent_transactions.rs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    /// Return the recently seen transactions
+    pub async fn get_all(&self) -> Result<Vec<FullTransaction>, Error> {
+        let (tx, rx) = oneshot::channel();
+
+        self.agent
+            .send(Commands::GetAll { resp: tx })
+            .await
+            .map_err(|_| Error::GoneOnSend)?;
+
+        rx.await.map_err(|_| Error::GoneOnRecv)
```
Comment (ineiti) [inline src/bin/server/recent_transactions.rs:74]: Same question here with regard to the channel.

File: tests/execs.rs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+use duct::cmd;
+use nix::sys::stat::{stat, Mode};
+
+const DIR: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/tests");
+
+const CLIENT_BIN: &str = env!("CARGO_BIN_EXE_client");
+const SERVER_BIN: &str = env!("CARGO_BIN_EXE_server");
+
+#[test]
+fn run_execs() {
```
Comment (ineiti) [inline tests/execs.rs:12]: Looks quite heavy to run the tests from rust - what is the advantage over running it from a shell?

**Review by ineiti — CHANGES_REQUESTED** [review]
Thanks for the overview in the PR - that makes it somewhat easier to check what's going on.

File: tests/execs.rs (hunk 4 above)
Comment (tharvik) [reply tests/execs.rs:12]: Discussed face2face: integrations tests are expected to be run by `cargo test`

**Review by ineiti — APPROVED** [review]

**Pull request merged**